yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala

```diff
@@ -231,6 +231,25 @@ private[spark] class ApplicationMaster(args: ApplicationMasterArguments,
     reporterThread = launchReporterThread()
   }
 
+  /**
+   * Creates an actor that ApplicationMaster communicates with YarnScheduler of driver
+   * in Yarn deploy mode.
+   *
+   * If isDriver is set to true, AMActor and driver belong to same process.
+   * so AMActor don't monitor lifecycle of driver.
+   */
+  private def runAMActor(
+      host: String,
+      port: String,
+      isDriver: Boolean): Unit = {
```

Comment:
can you rename `isDriver` to `isClusterMode`? It might be apparent to us that these two mean the same thing but for newcomers it's not obvious at all.